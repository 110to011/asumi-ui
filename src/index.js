/**
 * Created by Elly on 2016/5/27.
 */

import Menu from './component/menu';
import SubMenu from './component/menu/subMenu';
import MenuItem from './component/menu/menuItem';
import MenuItemGroup from './component/menu/menuItemGroup';
import Tag from './component/tag';
import Grid from './component/grid';
import Form from './component/form';
import Tabs from './component/tabs';
import Radio from './component/radio';
import Group from './component/group';
import Input from './component/input';
import Modal from './component/modal';
import Table from './component/table';
import Editor from './component/editor';
import Col from './component/table/col';
import Button from './component/button';
import Select from './component/select';
import Loading from './component/loading';
import Popover from './component/popover';
import Message from './component/message';
import Animate from './component/animate';
import Upload from './component/upload';
import Checkbox from './component/checkbox';
import Dropdown from './component/dropdown';
import Option from './component/select/option';
import Pagination from './component/pagination';
import FormItem from './component/form/formItem';
import TabPanel from './component/tabs/tabPanel';
import RadioGroup from './component/radio/radioGroup';
import CheckboxGroup from './component/checkbox/checkGroup';
import SimplePagination from './component/pagination/simplePagination';
import DateTime from './component/datetime';

const Tooltip = Popover;

Table.Col = Col;

Select.Option = Option;

Tabs.TabPanel = TabPanel;

Menu.SubMenu = SubMenu;
Menu.MenuItem = MenuItem;
Menu.MenuItemGroup = MenuItemGroup;

Radio.Group = RadioGroup;
Checkbox.Group = CheckboxGroup;

Form.FormItem = FormItem;

Pagination.Simple = SimplePagination;

export {
    Col,
    Tag,
    Tabs,
    Menu,
    Form,
    Grid,
    Table,
    Modal,
    Input,
    Group,
    Radio,
    Select,
    Button,
    Upload,
    Option,
    Editor,
    SubMenu,
    Animate,
    Loading,
    Popover,
    Message,
    Tooltip,
    FormItem,
    MenuItem,
    TabPanel,
    Dropdown,
    Checkbox,
    DateTime,
    Pagination,
    RadioGroup,
    CheckboxGroup,
    MenuItemGroup,
    SimplePagination,
};