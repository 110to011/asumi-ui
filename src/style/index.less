@charset "utf-8";

@import "variables";
@import "normalize";
@import "fontawesome/less/font-awesome";

body {
  font-family: '微软雅黑', "Microsoft YaHei", Arial, Microsoft YaHei, Helvetica, sans-serif;
  margin: 0;
  color: @normalText;
  font-size: @fontSize;
  line-height: 1.42857;
}

a {
  color: @primary;
  &:hover, &:active, &:visited, &:focus {
    color: @d_primary;
  }
}

.el-shadow {
  box-shadow: 0 6px 12px @shadow;
  -webkit-box-shadow: 0 6px 12px @shadow;
}

.overflow {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@import "nav";
@import "tabs";
@import "drop";
@import "caret";
@import "modal";
@import "input";
@import "table";
@import "button";
@import "stretch";
@import "popover";
@import "checkbox";
@import "pagination";
