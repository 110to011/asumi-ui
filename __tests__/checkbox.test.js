import React, {Component} from 'react';
import {shallow, mount} from 'enzyme';
import {Checkbox, CheckboxGroup} from '../src';

describe('Checkbox', () => {
    it('create component', () => {
        const component = mount(<Checkbox/>);
        expect(component).toMatchSnapshot();
    });
    it('define label', () => {
        const component = mount(<Checkbox label="label"/>);
        expect(component.text()).toEqual("label");
    });
    it('define children', () => {
        const component = mount(<Checkbox>label</Checkbox>);
        expect(component.text()).toEqual("label");
    });
    it('define checked', () => {
        const component = mount(<Checkbox checked/>);
        expect(component).toMatchSnapshot();
        expect(component.find('input[checked]').length).toEqual(1);
    });
    it('define disabled', () => {
        const component = mount(<Checkbox disabled/>);
        expect(component).toMatchSnapshot();
        expect(component.find('input[disabled]').length).toEqual(1);
    });
    it('define className', () => {
        const component = shallow(<Checkbox className="jest"/>);
        expect(component).toBeDefined();
        expect(component.find('.jest').length).toEqual(1);
    });
    it('define name', () => {
        const component = mount(<Checkbox name="jest"/>);
        expect(component).toBeDefined();
        expect(component.find('input[name="jest"]').length).toEqual(1);
    });
    it('define readOnly', () => {
        const component = mount(<Checkbox readOnly/>);
        expect(component).toMatchSnapshot();
        expect(component.find('input[readOnly]').length).toEqual(1);
    });
    it('define switch', () => {
        const component = mount(<Checkbox type="switch"/>);
        expect(component).toMatchSnapshot();
    });
    it('define onBlur', () => {
        const handleBlur = jest.fn();
        const component = mount(<Checkbox onBlur={handleBlur}/>);
        component.find('input').simulate('focus');
        component.find('input').simulate('blur');
        expect(handleBlur).toBeCalled();
    });
    it('define onKeyPress', () => {
        const handleKeyPress = jest.fn();
        const component = mount(<Checkbox onKeyPress={handleKeyPress}/>);
        component.find('input').simulate('keypress');
        expect(handleKeyPress).toBeCalled();
    });
    it('define onchange', () => {
        class Demo extends Component {
            constructor() {
                super();
                this.state = {
                    checked: false
                };
            }

            handleChange({value, checked}) {
                this.setState({checked})
            }

            render() {
                return (
                    <Checkbox
                        value={1}
                        ref={c => {
                            this._c = c
                        }}
                        checked={this.state.checked}
                        onChange={this.handleChange.bind(this)}/>)
            }
        }

        const component = mount(<Demo/>);
        component.find('input').simulate('change');
        expect(component.instance().state.checked).toEqual(true);
    });
});

describe("Checkbox Group", () => {
    it("create component", () => {
        const component = mount(
            <CheckboxGroup options={[{
                label: '选项1',
                value: 1
            }, {
                label: '选项2',
                value: 2
            }, {
                label: '选项3',
                value: 3
            }]}/>);
        expect(component).toMatchSnapshot();
    });
    it("define options str", () => {
        const component = mount(
            <CheckboxGroup
                disableAll
                options={[1, 2, 3]}/>);
        expect(component).toMatchSnapshot();
    });
    it("define disableAll", () => {
        const component = mount(
            <CheckboxGroup
                disableAll
                options={[{
                    label: '选项1',
                    value: 1
                }, {
                    label: '选项2',
                    value: 2
                }, {
                    label: '选项3',
                    value: 3
                }]}/>);
        expect(component).toMatchSnapshot();
    });
    it("define className", () => {
        const component = mount(
            <CheckboxGroup
                className="jest"
                options={[{
                    label: '选项1',
                    value: 1
                }, {
                    label: '选项2',
                    value: 2
                }, {
                    label: '选项3',
                    value: 3
                }]}/>);
        expect(component.find('.el-checkbox-group.jest').length).toEqual(1);
    });
    it('define onBlur', () => {
        const handleBlur = jest.fn();
        const component = mount(<CheckboxGroup
            onBlur={handleBlur}
            options={[{
                label: '选项1',
                value: 1
            }, {
                label: '选项2',
                value: 2
            }, {
                label: '选项3',
                value: 3
            }]}/>);
        component.find('input').at(0).simulate('focus');
        component.find('input').at(0).simulate('blur');
        expect(handleBlur).toBeCalled();
        component.find('input').at(2).simulate('focus');
        component.find('input').at(2).simulate('blur');
        expect(handleBlur).toBeCalled();
    });
    it('define onKeyPress', () => {
        const handleKeyPress = jest.fn();
        const component = mount(<CheckboxGroup
            onKeyPress={handleKeyPress}
            options={[{
                label: '选项1',
                value: 1
            }, {
                label: '选项2',
                value: 2
            }, {
                label: '选项3',
                value: 3
            }]}/>);
        component.find('input').at(0).simulate('keypress');
        expect(handleKeyPress).toBeCalled();
        component.find('input').at(2).simulate('keypress');
        expect(handleKeyPress).toBeCalled();
    });
    it("define hasCheckAll", () => {
        const component = mount(
            <CheckboxGroup
                hasCheckAll={false}
                options={[{
                    label: '选项1',
                    value: 1
                }, {
                    label: '选项2',
                    value: 2
                }, {
                    label: '选项3',
                    value: 3
                }]}/>);
        expect(component).toMatchSnapshot();
    });
    it('define onchange', () => {
        class Demo extends Component {
            constructor() {
                super();
                this.state = {
                    value: []
                };
            }

            handleChange({value}) {
                this.setState({value})
            }

            render() {
                return (
                    <CheckboxGroup
                        checkedList={this.state.value}
                        onChange={this.handleChange.bind(this)}
                        options={[{
                            label: '选项1',
                            value: 1
                        }, {
                            label: '选项2',
                            value: 2
                        }, {
                            label: '选项3',
                            value: 3
                        }]}/>)
            }
        }

        const component = mount(<Demo/>);
        component.find('.el-checkbox-row.el-check-all input').simulate('change');
        expect(component.instance().state.value).toEqual([1, 2, 3]);
        component.find('.el-checkbox-row.el-check-all input').simulate('change');
        expect(component.instance().state.value).toEqual([]);
    });
});


