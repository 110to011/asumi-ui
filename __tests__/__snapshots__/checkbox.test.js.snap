// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Group create component 1`] = `
<CheckGroup
  checkAllLabel="全选"
  checkedList={Array []}
  disableAll={false}
  hasCheckAll={true}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "选项1",
        "value": 1,
      },
      Object {
        "label": "选项2",
        "value": 2,
      },
      Object {
        "label": "选项3",
        "value": 3,
      },
    ]
  }
>
  <div
    className="el-checkbox-group"
  >
    <div
      className="el-checkbox-row el-check-all"
    >
      <Checkbox
        checked={false}
        checkedList={Array []}
        disabled={false}
        indeterminate={0}
        label="全选"
        onChange={[Function]}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={false}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            全选
          </span>
        </label>
      </Checkbox>
    </div>
    <div
      className="el-checkbox-row"
    >
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={false}
        key="0"
        label="选项1"
        onChange={[Function]}
        value={1}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
              value={1}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项1
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={false}
        key="1"
        label="选项2"
        onChange={[Function]}
        value={2}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
              value={2}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项2
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={false}
        key="2"
        label="选项3"
        onChange={[Function]}
        value={3}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
              value={3}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项3
          </span>
        </label>
      </Checkbox>
    </div>
  </div>
</CheckGroup>
`;

exports[`Checkbox Group define disableAll 1`] = `
<CheckGroup
  checkAllLabel="全选"
  checkedList={Array []}
  disableAll={true}
  hasCheckAll={true}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "选项1",
        "value": 1,
      },
      Object {
        "label": "选项2",
        "value": 2,
      },
      Object {
        "label": "选项3",
        "value": 3,
      },
    ]
  }
>
  <div
    className="el-checkbox-group"
  >
    <div
      className="el-checkbox-row el-check-all"
    >
      <Checkbox
        checked={false}
        checkedList={Array []}
        disabled={true}
        indeterminate={0}
        label="全选"
        onChange={[Function]}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={false}
              className="el-checkbox-input"
              disabled={true}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            全选
          </span>
        </label>
      </Checkbox>
    </div>
    <div
      className="el-checkbox-row"
    >
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={true}
        key="0"
        label="选项1"
        onChange={[Function]}
        value={1}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={true}
              onChange={[Function]}
              type="checkbox"
              value={1}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项1
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={true}
        key="1"
        label="选项2"
        onChange={[Function]}
        value={2}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={true}
              onChange={[Function]}
              type="checkbox"
              value={2}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项2
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={true}
        key="2"
        label="选项3"
        onChange={[Function]}
        value={3}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={true}
              onChange={[Function]}
              type="checkbox"
              value={3}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项3
          </span>
        </label>
      </Checkbox>
    </div>
  </div>
</CheckGroup>
`;

exports[`Checkbox Group define hasCheckAll 1`] = `
<CheckGroup
  checkAllLabel="全选"
  checkedList={Array []}
  disableAll={false}
  hasCheckAll={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "选项1",
        "value": 1,
      },
      Object {
        "label": "选项2",
        "value": 2,
      },
      Object {
        "label": "选项3",
        "value": 3,
      },
    ]
  }
>
  <div
    className="el-checkbox-group"
  >
    <div
      className="el-checkbox-row el-check-all"
    />
    <div
      className="el-checkbox-row"
    >
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={false}
        key="0"
        label="选项1"
        onChange={[Function]}
        value={1}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
              value={1}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项1
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={false}
        key="1"
        label="选项2"
        onChange={[Function]}
        value={2}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
              value={2}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项2
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={false}
        key="2"
        label="选项3"
        onChange={[Function]}
        value={3}
      >
        <label
          className="el-checkbox-wrapper"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
              value={3}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            选项3
          </span>
        </label>
      </Checkbox>
    </div>
  </div>
</CheckGroup>
`;

exports[`Checkbox Group define options str 1`] = `
<CheckGroup
  checkAllLabel="全选"
  checkedList={Array []}
  disableAll={true}
  hasCheckAll={true}
  onChange={[Function]}
  options={
    Array [
      1,
      2,
      3,
    ]
  }
>
  <div
    className="el-checkbox-group"
  >
    <div
      className="el-checkbox-row el-check-all"
    >
      <Checkbox
        checked={false}
        checkedList={Array []}
        disabled={true}
        indeterminate={0}
        label="全选"
        onChange={[Function]}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={false}
              className="el-checkbox-input"
              disabled={true}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            全选
          </span>
        </label>
      </Checkbox>
    </div>
    <div
      className="el-checkbox-row"
    >
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={true}
        key="0"
        label={1}
        name={1}
        onChange={[Function]}
        value={1}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={true}
              name={1}
              onChange={[Function]}
              type="checkbox"
              value={1}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            1
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={true}
        key="1"
        label={2}
        name={2}
        onChange={[Function]}
        value={2}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={true}
              name={2}
              onChange={[Function]}
              type="checkbox"
              value={2}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            2
          </span>
        </label>
      </Checkbox>
      <Checkbox
        checked={0}
        checkedList={Array []}
        disabled={true}
        key="2"
        label={3}
        name={3}
        onChange={[Function]}
        value={3}
      >
        <label
          className="el-checkbox-wrapper el-disabled"
        >
          <span
            className="el-checkbox"
          >
            <input
              checked={0}
              className="el-checkbox-input"
              disabled={true}
              name={3}
              onChange={[Function]}
              type="checkbox"
              value={3}
            />
            <span
              className="fa fa-square-o el-unchecked"
            />
          </span>
          <span>
            3
          </span>
        </label>
      </Checkbox>
    </div>
  </div>
</CheckGroup>
`;

exports[`Checkbox create component 1`] = `
<Checkbox
  checked={false}
  onChange={[Function]}
>
  <label
    className="el-checkbox-wrapper"
  >
    <span
      className="el-checkbox"
    >
      <input
        checked={false}
        className="el-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="fa fa-square-o el-unchecked"
      />
    </span>
    <span />
  </label>
</Checkbox>
`;

exports[`Checkbox define checked 1`] = `
<Checkbox
  checked={true}
  onChange={[Function]}
>
  <label
    className="el-checkbox-wrapper"
  >
    <span
      className="el-checkbox"
    >
      <input
        checked={true}
        className="el-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="fa fa-check-square el-checked"
      />
    </span>
    <span />
  </label>
</Checkbox>
`;

exports[`Checkbox define disabled 1`] = `
<Checkbox
  checked={false}
  disabled={true}
  onChange={[Function]}
>
  <label
    className="el-checkbox-wrapper el-disabled"
  >
    <span
      className="el-checkbox"
    >
      <input
        checked={false}
        className="el-checkbox-input"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="fa fa-square-o el-unchecked"
      />
    </span>
    <span />
  </label>
</Checkbox>
`;

exports[`Checkbox define readOnly 1`] = `
<Checkbox
  checked={false}
  onChange={[Function]}
  readOnly={true}
>
  <label
    className="el-checkbox-wrapper"
  >
    <span
      className="el-checkbox"
    >
      <input
        checked={false}
        className="el-checkbox-input"
        onChange={[Function]}
        readOnly={true}
        type="checkbox"
      />
      <span
        className="fa fa-square-o el-unchecked"
      />
    </span>
    <span />
  </label>
</Checkbox>
`;

exports[`Checkbox define switch 1`] = `
<Checkbox
  checked={false}
  onChange={[Function]}
  type="switch"
>
  <label
    className="el-checkbox-wrapper"
  >
    <span
      className="el-checkbox el-switch"
    >
      <input
        checked={false}
        className="el-checkbox-input"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
    </span>
    <span />
  </label>
</Checkbox>
`;
